lib/wallet/cardano-wallet.cabal

```diff
@@ -127,6 +127,7 @@ library
     , lifted-async
     , math-functions
     , memory
+    , mmorph
```

Comment:
```suggestion
```

I would be in favor of keeping our dependency footprint small, especially since this code is temporary. 🤔   We can use `Control.Monad.Trans.Reader.mapReaderT` which specializes `hoist` to `ReaderT`.

Reply:
I think the hoist concept is already in use in the repo and quite  useful in the mtl style. I would leave it if you do not mind

Reply:
I would indeed strongly favor not adding `mmorph`. 😅 The reasons are twofold: I tend to be conservative when importing packages (compile times, version bumps, CI …) and to apply a fairly high [Fairbairn threshold](https://wiki.haskell.org/Fairbairn_threshold). In this case, `mapReaderT` from `transformers` is a readily available specialization of `hoist`, so I'd settle for this one.

The other reason is that I feel that the `mmorph` packages doesn't quite capture the "right" point in the design space. It's true that we do have a couple of functions named `hoist*` in the codebase, e.g.

* `hoistLightSyncSource`
* `hoistTimeInterpreter`
* `hoistEraValue`
* `hoistStats`
* … and more

However, of the functions that I mentioned here, actually _none_ of them is a specialization of the `hoist` function from the `mmorph` package — the reason is that the `mmorph` packages works with monads `∀m. Monad m => m a -> n a`, while the hoist-functions above work with general natural transformations `∀a. f a -> g a`. In other words, the `mmorph` package only captures a subset of our use cases.

That said, I'm totally on-board with calling everything that “maps” a natural transformation `∀a. f a → g a` a “hoist”. My issue is not with the concept, just with the marginal value of adding the package dependency.